ibis/sql/postgres/compiler.py

```diff
@@ -622,3 +637,26 @@ class PostgreSQLExprTranslator(alch.AlchemyExprTranslator):
 class PostgreSQLDialect(alch.AlchemyDialect):
 
     translator = PostgreSQLExprTranslator
+
+
+@P.upcast.register(
+    PostgreSQLExprTranslator,
+    (ops.Sum, ops.Mean, ops.Min, ops.Max),
+    ir.BooleanColumn,
+    (ir.BooleanColumn, type(None))
+)
+def upcast_simple_boolean_reduction(_, operation, column, where):
+    return type(operation)(column.cast('int32'), where).to_expr()
+
+
+@P.upcast.register(
+    PostgreSQLExprTranslator,
+    (ops.StandardDev, ops.Variance),
+    ir.BooleanColumn,
+    (ir.BooleanColumn, type(None)),
+    six.string_types
+)
+def upcast_variance_stddev_boolean_reduction(
+    _, operation, column, where, kind
+):
+    return type(operation)(column.cast('int32'), where, kind).to_expr()
```

Comment:
I don't have a strong view, but you like this more than putting the casting logic into the translator implementations for these operators? I guess if you start accumulating a lot of upcasting rules then this is a little easier